pkg/datapath/loader/netlink.go

```diff
@@ -88,8 +89,24 @@ func replaceDatapath(ctx context.Context, ifName, objPath, progName, direction s
 		return nil, fmt.Errorf("loading eBPF ELF: %w", err)
 	}
 
-	if spec.Programs[progName] == nil {
-		return nil, fmt.Errorf("no program %s found in eBPF ELF", progName)
+	progsToAttach := make([]ProgDefinition, 0, len(progs))
+	links := make([]netlink.Link, 0, len(progs))
+	var errs error
+
+	for _, prog := range progs {
+		link, err := netlink.LinkByName(prog.ifName)
+		if err != nil {
+			errs = multierr.Append(errs, fmt.Errorf("getting interface %s by name: %w", prog.ifName, err))
```

Comment:
Sure, but `replaceDatapath` is intended to be used within the scope of a single Endpoint for now, so if one prog fails to attach, there's no point in going ahead with the other. Let's keep the logic simple.

Reply:
Sorry to keep pushing this, I just want to be sure that the logic remains the same more or less and I won't break anything. As I see in `reloadDatapath`, we invoke `replaceDatapath` for ingress and egress programs separately, so we can successfully attach the ingress program, and then fail on egress. If we fail immediately on updated `replaceDatapath`, then we won't be able to successfully attach ingress program. Probably, that's the intended behavior in this case, but I want to be sure about it.

Reply:
By `ingress and egress programs separately`, do you mean we attach the egress program conditionally? Regardless, if one of the endpoint's programs fail to attach, we can to give up immediately, since it means the CNI attach operation needs to be aborted and retried from the start. I don't see a reason to try both, and that's how the current setup works, if I understand correctly.

Reply:
I see, makes sense, thanks for the details!